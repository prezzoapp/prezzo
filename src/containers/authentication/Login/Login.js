// @flow
import React from 'react';
import { Header } from 'react-navigation';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  KeyboardAvoidingView,
  ScrollView,
  Platform,
  NetInfo,
  ImageBackground
} from 'react-native';
import PropTypes from 'prop-types';
import { heightPercentageToDP as hp, widthPercentageToDP as wp } from 'react-native-responsive-screen';
import {
  FONT_FAMILY,
  FONT_FAMILY_REGULAR,
  FONT_FAMILY_MEDIUM
} from '../../../services/constants';
import { Feather } from '../../../components/VectorIcons';
import LoginTextInput from '../../../components/LoginTextInput';
import Button from '../../../components/Button';
import { Constants } from 'expo';
import LoadingComponent from '../../../components/LoadingComponent';
import CacheImage from '../../../components/CacheImage';
import { showAlertWithMessage } from '../../../services/commonFunctions';

type Props = {
  loginWithEmail: Function,
  navigate: PropTypes.func.isRequired
};

type State = {
  email: string,
  password: string
};

const SCROLL_VIEW_TOP_PADDING = hp('13.42%') - (Header.HEIGHT + Constants.statusBarHeight - (Platform.OS === 'ios' ? 13 : 0));

class Login extends React.Component<Props, State> {
  static navigationOptions = ({ navigation }) => ({
    headerStyle: {
      position: 'absolute',
      backgroundColor: 'transparent',
      zIndex: 100,
      top: 0,
      left: 0,
      right: 0,
      borderBottomColor: 'transparent',
      elevation: 0
    },
    headerTintColor: '#fff',
    headerLeft: (
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => navigation.goBack()}
        style={styles.headerLeftBtn}>
        <Feather
          title="Back"
          name="chevron-left"
          color="white"
          size={wp('8%')}
        />
      </TouchableOpacity>
    )
  });

  state = {
    email: '',
    password: ''
  };

  navigateToSignup() {
    this.props.navigate({ routeName: 'SignupName' });
  }

  navigateToMain() {
    this.props.navigate({ routeName: 'Customer' });
  }

  afterLogin() {
    this.navigateToMain();
  }

  login() {
    const { email, password } = this.state;

    this.props.loginWithEmail(email, password)
      .then(() => this.afterLogin())
    .catch(e => showAlertWithMessage('Uh-oh!', e));
  }

  render() {
    const { email, password } = this.state;
    return (
      <CacheImage
        style={styles.container}
        type='backgroundImage'
        source={require('../../../../assets/images/bg/authentication.jpg')}
      >
        <KeyboardAvoidingView
          style={{ flex: 1 }}
          behavior='padding'>
          <ScrollView
            keyboardShouldPersistTaps="handled"
            contentContainerStyle={styles.scrollView}>
            <Text testID="welcomeText" style={styles.headerText}>
              Welcome back!
            </Text>

            <Text testID="signinText" style={styles.subHeaderText}>
              Sign In To Continue
            </Text>

            <LoginTextInput
              type="email"
              label="Email Address"
              value={email}
              onChange={email => this.setState({ email })}
            />

            <LoginTextInput
              type="password"
              label="Password"
              value={password}
              onChange={password => this.setState({ password })}
            />

            <TouchableOpacity
              testID="signupButton"
              style={styles.signupLabelContainer}
              onPress={() => this.navigateToSignup()}
            >
              <Text style={styles.signupLabel}>Don't have an account?</Text>

              <Text style={styles.signupLink}>Sign Up</Text>
            </TouchableOpacity>

            <View style={styles.buttonContainer}>
              <Button
                style={buttonStyles.login}
                textStyle={buttonStyles.loginText}
                onPress={() => this.login()}
              >
                Sign In
              </Button>
            </View>
          </ScrollView>
        </KeyboardAvoidingView>
        <LoadingComponent visible={this.props.isBusy} />
      </CacheImage>
    );
  }
}

const containerPaddingLeftRight: number = wp('10.66%');
const containerPaddingTopBottom: number = hp('9.85%');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#4A4A4A',
    paddingTop: Header.HEIGHT + Constants.statusBarHeight - (Platform.OS === 'ios' ? 20 : 0)
  },
  scrollView: {
    paddingLeft: containerPaddingLeftRight,
    paddingRight: containerPaddingLeftRight,
    paddingTop: SCROLL_VIEW_TOP_PADDING,
    paddingBottom: hp('5%')
  },
  headerText: {
    fontSize: wp('10.16%'),
    fontFamily: FONT_FAMILY_MEDIUM,
    color: '#fff',
    letterSpacing: 0,
    backgroundColor: 'transparent',
    paddingTop: hp('0%')
  },
  subHeaderText: {
    fontSize: wp('6.13%'),
    fontFamily: FONT_FAMILY,
    color: '#fff',
    letterSpacing: 0,
    marginBottom: hp('4.92%'),
    backgroundColor: 'transparent',
    marginTop: 5
  },
  signupLabelContainer: {
    width: '100%',
    height: 'auto',
    marginBottom: wp('17.33%'),
    marginTop: wp('10.66%'),
    backgroundColor: 'transparent',
    flexDirection: 'row',
    display: 'flex'
  },
  signupLabel: {
    color: '#959595',
    fontSize: wp('4.26%'),
    fontFamily: FONT_FAMILY,
    marginRight: 4
  },
  signupLink: {
    color: '#fff',
    fontSize: wp('4.53%'),
    fontFamily: FONT_FAMILY
  },
  buttonContainer: {
    alignItems: 'center'
  },
  headerLeftBtn: {
    marginLeft: wp('4.4%')
  }
});

const buttonStyles = {
  login: {
    width: wp('42.66%'),
    height: hp('6.65%'),
    backgroundColor: '#0DD24A',
    borderColor: '#0DD24A',
    justifyContent: 'center'
  },
  loginText: {
    fontSize: wp('5.33%'),
    fontFamily: FONT_FAMILY_MEDIUM,
    paddingTop: 0,
    paddingBottom: 0
  }
};

export default Login;
