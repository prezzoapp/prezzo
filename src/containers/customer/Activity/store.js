// @flow
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';

const mapDispatchToProps = dispatch => {
  return {
    navigate: bindActionCreators(NavigationActions.navigate, dispatch)
  };
};

export default mapDispatchToProps;
