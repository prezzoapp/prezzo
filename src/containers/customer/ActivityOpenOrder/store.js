import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';
import { listOpenOrders } from '../../../modules/userActivity';

export const mapStateToProps = state => {
  const data = state
    .get('userActivity')
    .get('data')
    .toJS();

  const isBusy = state.get('userActivity').get('isBusy');

  return {
    data,
    isBusy
  };
};

export const mapDispatchToProps = dispatch => {
  return {
    listOpenOrders: bindActionCreators(listOpenOrders, dispatch),
    navigate: bindActionCreators(NavigationActions.navigate, dispatch)
  };
};
