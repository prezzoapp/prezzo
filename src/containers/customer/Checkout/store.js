// @flow
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';

import { addRemoveItemQuantity, setType } from '../../../modules/restaurant';
import { listCreditCards } from '../../../modules/paymentMethods';
import { userLogout } from '../../../modules/auth';

export const mapStateToProps = state => {
  const data = state.get('restaurant').toJS();
  const type = state.get('restaurant').get('type');
  const paymentType = state.get('restaurant').get('paymentType');
  const newlyAddedCard = state.get('paymentMethods').get('newlyAddedCard');
  const creditCardList = state
    .get('paymentMethods')
    .get('data')
    .toJS();

  return {
    data,
    type,
    paymentType,
    creditCardList,
    newlyAddedCard
  };
};

export const mapDispatchToProps = dispatch => ({
  navigate: bindActionCreators(NavigationActions.navigate, dispatch),
  addRemoveItemQuantity: bindActionCreators(addRemoveItemQuantity, dispatch),
  setType: bindActionCreators(setType, dispatch),
  listCreditCards: bindActionCreators(listCreditCards, dispatch),
  userLogout: bindActionCreators(userLogout, dispatch)
});
