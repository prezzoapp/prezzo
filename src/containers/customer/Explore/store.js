// @flow
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';
import { listVendors, toggleFilter, updateDistance, updatePrice, getUserCurrentLocation, callWaiterBtnFunc } from '../../../modules/explore';

export const mapStateToProps = state => {
  const filters = state
    .get('explore')
    .get('filters')
    .toJS();
  const distance = state.get('explore').get('distance');
  const pricing = state.get('explore').get('pricing');
  const isBusy = state.get('explore').get('isBusy');
  const minDistance = state.get('explore').get('minDistance');
  const maxDistance = state.get('explore').get('maxDistance');
  const callWaiterBtnState = state.get('explore').get('callWaiterBtnState')
  const userName = state.get('user').get('account').get('fullName');
  const isUserOpenOrder = state.get('userActivity').get('data').size !== 0 ? true : false

  return {
    filters,
    distance,
    pricing,
    isBusy,
    minDistance,
    maxDistance,
    userName,
    callWaiterBtnState,
    isUserOpenOrder
  };
};

export const mapDispatchToProps = dispatch => ({
  navigate: bindActionCreators(NavigationActions.navigate, dispatch),
  listVendors: bindActionCreators(listVendors, dispatch),
  toggleFilter: bindActionCreators(toggleFilter, dispatch),
  updateDistance: bindActionCreators(updateDistance, dispatch),
  updatePrice: bindActionCreators(updatePrice, dispatch),
  getUserCurrentLocation: bindActionCreators(getUserCurrentLocation, dispatch),
  callWaiterBtnFunc: bindActionCreators(callWaiterBtnFunc, dispatch)
});
