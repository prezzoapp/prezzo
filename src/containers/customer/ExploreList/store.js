// @flow
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';
import { addRestaurantDetail } from '../../../modules/restaurant';

export const mapStateToProps = state => {
  const explore = state.get('explore');
  const restaurants = explore.get('restaurants').toJS();

  return {
    restaurants
  };
};

export const mapDispatchToProps = dispatch => ({
  navigate: bindActionCreators(NavigationActions.navigate, dispatch),
  addRestaurantDetail: bindActionCreators(addRestaurantDetail, dispatch)
});
