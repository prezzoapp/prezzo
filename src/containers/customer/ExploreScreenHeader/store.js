// @flow
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';
import { toggleFilter, updateDistance } from '../../../modules/explore';

export const mapStateToProps = state => {
  const filters = state
    .get('explore')
    .get('filters')
    .toJS();
  const minDistance = state.get('explore').get('minDistance');
  const maxDistance = state.get('explore').get('maxDistance');
  const distance = state.get('explore').get('distance');

  return {
    filters,
    minDistance,
    maxDistance,
    distance
  };
};

export const mapDispatchToProps = dispatch => ({
  navigate: bindActionCreators(NavigationActions.navigate, dispatch),
  toggleFilter: bindActionCreators(toggleFilter, dispatch),
  updateDistance: bindActionCreators(updateDistance, dispatch)
});
