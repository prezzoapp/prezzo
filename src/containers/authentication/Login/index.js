// @flow
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';
import { loginWithEmail } from '../../../modules/auth';
import { listCreditCards } from '../../../modules/paymentMethods';
import Login from './Login';

const mapStateToProps = state => ({
  isBusy: state.get('auth').get('isBusy'),
});

export default connect(
  mapStateToProps,
  dispatch => {
    return {
      loginWithEmail: bindActionCreators(loginWithEmail, dispatch),
      listCreditCards: bindActionCreators(listCreditCards, dispatch),
      navigate: bindActionCreators(NavigationActions.navigate, dispatch)
    };
  }
)(Login);
