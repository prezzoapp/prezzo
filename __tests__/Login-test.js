import 'react-native';
import React from 'react';
import Login from '../src/containers/authentication/Login';
import store from '../src/redux/store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

it('Tourial Screen renders correctly', () => {
  const login = renderer
    .create(
      <Provider store={store}>
        <Login />
      </Provider>
    )
    .toJSON();
  expect(login).toMatchSnapshot();
});
