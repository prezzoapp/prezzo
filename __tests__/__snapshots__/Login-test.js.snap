// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tourial Screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#4A4A4A",
      "flex": 1,
      "paddingBottom": 80,
      "paddingLeft": 40,
      "paddingRight": 40,
      "paddingTop": 80,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 30,
      }
    }
    testID="welcomeText"
  >
    Welcome back!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 24,
        "marginBottom": 40,
      }
    }
    testID="signinText"
  >
    Sign In To Continue
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "#0DD24A",
        "borderBottomWidth": 2,
        "height": "auto",
        "marginBottom": 40,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": "auto",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#959595",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "600",
            "paddingBottom": 10,
          }
        }
      >
        Email Address
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      secureTextEntry={false}
      style={
        Object {
          "color": "white",
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "paddingBottom": 10,
        }
      }
      testID="loginTextInput"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "#0DD24A",
        "borderBottomWidth": 2,
        "height": "auto",
        "marginBottom": 40,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": "auto",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#959595",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "600",
            "paddingBottom": 10,
          }
        }
      >
        Password
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#959595",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "600",
              "paddingBottom": 10,
            }
          }
        >
          Show
        </Text>
      </View>
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        Object {
          "color": "white",
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "paddingBottom": 10,
        }
      }
      testID="loginTextInput"
      value=""
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "display": "flex",
        "flexDirection": "row",
        "height": "auto",
        "marginBottom": 40,
        "opacity": 1,
        "width": "100%",
      }
    }
    testID="signupButton"
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#959595",
          "fontFamily": "System",
          "fontSize": 16,
          "marginRight": 4,
        }
      }
    >
      Don't have an account?
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
    >
      Sign Up
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "width": "50%",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0DD24A",
          "borderColor": "#0DD24A",
          "borderRadius": 10,
          "borderWidth": 2,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="buttonComponent"
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontFamily": "System",
            "fontSize": 20,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Sign In
      </Text>
    </View>
  </View>
</View>
`;
